packages/expo-cli/src/commands/eas-build/build/utils/git.ts

```diff
@@ -36,27 +37,60 @@ async function showDiff(): Promise<void> {
   await spawnAsync('git', ['--no-pager', 'diff'], { stdio: ['ignore', 'inherit', 'inherit'] });
 }
 
-async function commitChangesAsync(commitMessage?: string): Promise<void> {
-  if (!commitMessage) {
-    const promptResult = await prompts({
-      type: 'text',
-      name: 'message',
-      message: 'Commit message:',
-      initial: 'Configure Xcode project',
-      validate: input => input !== '',
+async function addFile(file: string, options?: { intentToAdd?: boolean }): Promise<void> {
+  if (options?.intentToAdd) {
+    await spawnAsync('git', ['add', '--intent-to-add', file]);
+  } else {
+    await spawnAsync('git', ['add', file]);
+  }
+}
+
+async function reviewAndCommitChangesAsync(
+  commitMessage: string,
+  { nonInteractive }: { nonInteractive: boolean }
+): Promise<void> {
+  let message = commitMessage;
+  let shouldCommit = true;
```

Comment:
We shouldn't commit those changes if it's a non-interactive mode, in most cases, it would be run on ci where those committed changes would be lost. In case there is some changes in non-interactive mode we should throw an exception